scripts/gold_accounting.py

```diff
@@ -0,0 +1,352 @@
+        elif self.trans_id.startswith('Sch_'):
+            return 'Stripe'
+        elif self.trans_id.startswith('C'):
+            return 'Coinbase'
+        
+        return 'Unknown'
+
+    @property
+    def revenue(self):
+        if 'pennies' in self.overrides:
+            return self.overrides['pennies']
+
+        return self.pennies
+
+    @property
+    def months(self):
+        if self.days % 366 == 0:
+            return self.days / 366 * 12
+        elif self.days % 31 == 0:
+            return self.days / 31
+        else:
+            raise ValueError
+
+    @classmethod
+    def get_transactions(cls, start_date=None, end_date=None):
+        query = session.query(cls)
+        if start_date:
+            query = query.filter(cls.date >= start_date)
+        if end_date:
+            query = query.filter(cls.date < end_date)
+        
+        query = query.order_by(cls.date, cls.trans_id)
+        transactions = query.all()
+
+        overrides = {}
+        for trans in transactions:
+            if overrides or not trans.is_split:
+                trans.overrides = overrides
+                overrides = {}
+                yield trans
+            else:
+                overrides['type'] = trans.type
+                overrides['pennies'] = trans.pennies
+                overrides['trans_id'] = trans.trans_id
+                overrides['processor'] = trans.processor
+                overrides['description'] = trans.description
+                overrides['is_reversed'] = trans.is_reversed
+
+
+class Creddit(object):
+    def __init__(self, value, processor, purchase_time):
+        self.value = value
+        self.processor = processor
+        self.purchase_time = purchase_time
+        if self.purchase_time < datetime(2013, 1, 1, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=-420, name=None)):
+            self.value = 0
+
+
+def get_creddit_balances_on(date):
+    user_creddits = defaultdict(deque)
+    transactions = GoldTransaction.get_transactions(end_date=date)
+    for trans in transactions:
+        if trans.type == 'creddits':
+            creddit = Creddit(value=trans.revenue / trans.months,
+                              processor=trans.processor,
+                              purchase_time=trans.date)
+            user_creddits[trans.account_id].extend([creddit] * trans.months)
+        elif trans.type.endswith('-creddit'):
+            for _ in xrange(trans.months):
+                try:
+                    user_creddits[trans.payer].popleft()
+                except:
+                    break
+
+    return user_creddits
+
+
+def run_for_prev_month():
+    end_date = datetime(datetime.now().year, datetime.now().month, 1)
+    if end_date.month == 1:
+        start_date = datetime(end_date.year-1, 12, 1)
+    else:
+        start_date = datetime(end_date.year, end_date.month-1, 1)
+
+    user_creddits = get_creddit_balances_on(start_date)
+
+    output = cStringIO.StringIO()
+    output.write('"{}"'.format(
```

Comment:
Are you able to use `csv.writer(StringIO.StringIO(output))` here instead? Should give you much the same interface as `csvfile`. You might also have to use `StringIO` rather than `cStringIO` in that case as well, `cStringIO` really doesn't like unicode strings.